### ISSUE TITLE
Translation preview tool

### ISSUE THREAD

**Issue by Blueberryy** [Whimsyduke/SC2_GameTranslater#4]
WIth this feature you will be able to preview text in a window that will 'simulate' map game's UI. In every places where that string shows. You will need to set game's path to run it.
It might be very useful to let translator know context of the text if user has map source files to see how it will look.

It may also be called live-translation.

**Comment by Whimsyduke** [comment]
I apologize for taking so long to respond. I've been out of the StarCraft II map editor loop for a long time for a number of reasons. And rarely pay attention to the news on GitHub.
I can't clearly understand the meaning of “simulate' map game's UI”. Would you like a preview of how the translation will look like when the text is displayed on the StarCraft II UI via the “Text Message” action? If so, then it might be a bit of a pain in the ass, since SE's text support includes a number of features such as displaying images.
I don't think I have the time or inclination to do that at the moment. I used to work on the Warcraft 3 recast as an employee of NetEase. There was a big chunk of editor functionality (making the World Editor's trigger editor work similarly to the Starcraft 2 editor), but due to changes in the Classic group, it didn't end up being possible to get the results of over a year's worth of my endless overtime work to go live. It made me examine whether my own dedication to Blizzard was worth it.
I've probably come up with an alternative, I could generate a copy of the Galaxy code, call the “Create Cheat” action to display it in the Tips, and you could copy that part of the code into a script and go to Help[F12] in the Tips to see the results.For exampleLike SummerFor exampleHere's the imageFor exampleYou can copy this part of the code into a script and go to Help[F12] Tips to see the result as shown in the following figure.

![Image](https://github.com/user-attachments/assets/42e83ea0-1194-4bfe-a08d-6850228e59be)

**Comment by Blueberryy** [comment]
> I've probably come up with an alternative, I could generate a copy of the Galaxy code, call the “Create Cheat” action to display it in the Tips, and you could copy that part of the code into a script and go to Help[F12] in the Tips to see the results.For exampleLike SummerFor exampleHere's the imageFor exampleYou can copy this part of the code into a script and go to Help[F12] Tips to see the result as shown in the following figure.
> 
> ![Image](https://github.com/user-attachments/assets/42e83ea0-1194-4bfe-a08d-6850228e59be)

If that will work in your program that might be useful 😄 I'm not a developer/map author so i don't know how that would work so if it will be doing complete different things to display stuff so please win in this war or coding!

**Comment by Whimsyduke** [comment]
It looks like Blizzard has made some changes that result in only being able to launch SC2 by Battle.Net, in my test, launching SC2 via the command line fails. The earlier effective solution for launching SC2 is currently unavailable. Presumably BN has done some processing via shared memory.
So the most desirable solution has to be abandoned. The only thing I can do at the moment is to modify a map and then the user loads the map themselves with the editor and then runs the test.
I just don't even like this solution myself. I still want the translation tool to be able to start SC2 in debug mode (similar to test doc in Editor.) I don't have enough time to do further research since Chinese New Year is coming soon (need to reverse engineer BN, not something I can handle in a short time).
This will have to be put on hold for now. Or I can wait until I have time after the Chinese New Year to get a version that generates Galaxy code first, and then the user fills in their own custom script for that map (it will be as string to text instead of writing directly to text, because writing directly to a file will cause the editor to reload the map), and then debug the map and preview the text. Lucky i had made a tool for convert unicode character.

**Comment by Blueberryy** [comment]
Is it possible to make like this?
![Image](https://github.com/user-attachments/assets/c2fa5ba5-8c01-492a-92f0-f26c92031428)

**Comment by Whimsyduke** [comment]
No. Because text of SC2 can show pricture or compute result or show the data of game. If making a preview same as game in the translater, I need add a lot of code for reading and parsing files from map. So the best way is preview in game. 
I having finished the work now and going to upload .

**Comment by Whimsyduke** [comment]
@Blueberryy See https://github.com/Whimsyduke/SC2_GameTranslater/releases/tag/v1.1.0
